### DISCUSSION TITLE
help me with image describe option

### DISCUSSION THREAD

**Discussion by robotics-dpsbksc** [lllyasviel/Fooocus#2876] [Q&A] [answered]
i am using Fooocus on google Collab. Now i want to using the api from gradio to my python program, where it uses images from my computer, gives the image to the API and describes it to generate a prompt

**Comment by mashb1t** [comment, marked as answer]
Hey @robotics-dpsbksc,

You can find the API definition in the footer of Fooocus, you just have to get the correct fn_index (i assume it's 53)

```
from gradio_client import Client

client = Client("https://fooocus.mydomain.com/")
result = client.predict(
				"Photograph",	# str in 'Content Type' Radio component
				"https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png",	# str (filepath or URL to image)
								in 'Drag any image to here' Image component
				fn_index=53
)
print(result)
```

**Reply by robotics-dpsbksc** [reply]
(env) C:\Users\danygamer\Desktop\fkteck>python itot.py
Loaded as API: https://1d00db5ff8fb4a2d3d.gradio.live/ ✔
Traceback (most recent call last):
  File "C:\Users\danygamer\Desktop\fkteck\itot.py", line 4, in <module>
    result = client.predict(
             ^^^^^^^^^^^^^^^
  File "C:\Users\danygamer\AppData\Roaming\Python\Python312\site-packages\gradio_client\client.py", line 454, in predict
    inferred_fn_index = self._infer_fn_index(api_name, fn_index)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\danygamer\AppData\Roaming\Python\Python312\site-packages\gradio_client\client.py", line 787, in _infer_fn_index
    raise ValueError(f"Invalid function index: {fn_index}.")
ValueError: Invalid function index: 53.

**Reply by mashb1t** [reply]
you'd have to check yourself which index it is for you, might also be 54. Please check the footer where you can find the api definition.

**Reply by robotics-dpsbksc** [reply]
thanks man i searched for this whole time but i just need one more help i want the image from my computer to be used instead of images on internet

**Reply by mashb1t** [reply]
You can send the image as base64 encoded string instead of the URL. Please find an example in https://github.com/lllyasviel/Fooocus/discussions/2608

**Reply by robotics-dpsbksc** [reply]
its good but like i am using fooocus on google collab and want to use picture from my computer

**Reply by mashb1t** [reply]
then you need to upload the image to Colab first or mount your Google Drive in Colab to use your images saved on Google Drive there

**Reply by robotics-dpsbksc** [reply]
you told something about base64 will that work ?

**Reply by mashb1t** [reply]
A base64 encoded image is the image in text form as a string, so the whole image is transmitted without the need to upload it anywhere manually.
Based on your response i assumed that you've already tried this option. It's the most flexible way of providing an image to an API for sure.

**Reply by robotics-dpsbksc** [reply]
yeh it worked great! thanks man for all the support
